niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/SignaturePermissionTester.java

```diff
@@ -3656,18 +3875,625 @@ public void writeToParcel(Parcel parcel, int i) {
+                    builder = (KeyGenParameterSpec.Builder)invokeReflectionCall(builder.getClass(),
+                            "setUniqueIdIncluded",builder,
+                            new Class<?>[]{boolean.class}, true);
+
+                    KeyGenParameterSpec spec = builder.build();
+                    KeyStore keyStore = null;
+                    try {
+                        keyStore = KeyStore.getInstance("AndroidKeyStore");
+                        keyStore.load(null);
+                        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance(KEY_ALGORITHM_EC,
+                                "AndroidKeyStore");
+                        keyPairGenerator.initialize(spec);
+                        keyPairGenerator.generateKeyPair();
+                    } catch (java.security.ProviderException e){
+                        //If permission is not granted the module raise : java.security.ProviderException: Failed to generate key pair.
+                        throw new SecurityException(e);
+                    } catch (KeyStoreException | CertificateException | IOException | NoSuchAlgorithmException | InvalidAlgorithmParameterException | NoSuchProviderException e) {
+                        throw new UnexpectedPermissionTestFailureException(e);
+                    }
+                }));
+
+        // # Skip GET_HISTORICAL_APP_OPS_STATS
+        // Reason : The corresponding api hasn't raise any exception internally.(AppOpsService#getHistoricalOps)
+
+        mPermissionTasks.put(permission.SET_SYSTEM_AUDIO_CAPTION,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+                    invokeReflectionCall(mCaptioningManager.getClass(),
+                            "setSystemAudioCaptioningEnabled",mCaptioningManager,
+                            new Class<?>[]{boolean.class}, true);
+                }));
+
+
+        // # Skip INSTALL_DPC_PACKAGES
+        // Reason : Found no way to evaluate this permission without enterprise environment
+
+        //The Permission required for CTS test - Notification test suite - need platform signature to run
+
+//        mPermissionTasks.put(permission.REVOKE_POST_NOTIFICATIONS_WITHOUT_KILL,
+//                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+//                    @SuppressLint("WrongConstant") Object permissionManager = mContext.getSystemService("permission");
+//                    //revokePostNotificationPermissionWithoutKillForTest( java.lang.String int);
+//                     invokeReflectionCall(permissionManager.getClass(),
+//                            "revokePostNotificationPermissionWithoutKillForTest",permissionManager,
+//                            new Class<?>[]{String.class,int.class}, mContext.getPackageName(),0);
+//                }));
+
+
+        // # Skip DELIVER_COMPANION_MESSAGES
+        //Reason : This feature appears to have been punted to U. See b/199427116,
```

Comment:
Let's remove reference to this feature and the corresponding bug; if no APIs are available in the current release to test the permission, let's use something similar to your message above: `Reason : Found No corresponding API or implementation to test.`

Reply:
I see

Reply:
fixed

Reply:
Thanks!